rails-commenting.rb

```diff
@@ -7,24 +7,30 @@
 # FILE: app/controller/blog_posts_controller.rb
 
 # ---1)
+# BlogPosts controller directs the flow of the application.
+  #The controller collects information from database, stores local variable inside methods, and responsible for displaying the views.
 class BlogPostsController < ApplicationController
   def index
     # ---2)
+    #Method to all the blog post landing page
     @posts = BlogPost.all
   end
 
   # ---3)
+   #Local variable, custom made method that will send back one instance of this model being queried. For example, Only blogpost that has an id will be displayed.Find Id that is passed through form or URL
   def show
     @post = BlogPost.find(params[:id])
   end
 
   # ---4)
+  # Method to post a new blogpost and send that to the database.
```

Comment:
New and Edit are methods that have a sole purpose to setup a form view.  Create and Update will actually provide content to the database.